src/App.js

```diff
@@ -1,27 +1,73 @@
 import './App.css';
 import React from 'react';
 import SearchAndFilter from './components/SearchAndFilter';
+import jwt_decode from "jwt-decode";
 import 'bootstrap/dist/css/bootstrap.min.css';
+import {Button, Dropdown, DropdownButton}  from "react-bootstrap";
 
 class App extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      data: null
+      token: null,
+      user: {}
     };
   }
 
+  handleCallbackResponse = (response) => {
+    this.setState({token: response.credential});
+    let userObject = jwt_decode(response.credential);
+    this.setState({user: userObject});
+    document.getElementById("signInDiv").hidden = true;
+  }
+
+  handleLogout = () => {
+    this.setState({user: {}});
+    document.getElementById("signInDiv").hidden = false;
```

Comment:
hidden is not as powerful as display, and will be overriden by display.

Prefer to have a CSS class be like 
.hide {
  display: none;
}
and use classlist to apply/remove it rather than setting hidden = true/false